lib/utils/has-credentials.js

```diff
@@ -0,0 +1,5 @@
+function hasCredentials (creds) {
```

Comment:
The only thing this is ever passed is `getCredentialsById`. This really feels like it should be either an attribute on the return object of that function, or another helper function inside config that is also passed the registry.  I would argue for the former, given the use case of the function i.e. what we're really checking.

I would say that moving this into a separate function might even be premature, and this is a time when it's ok to repeat ourselves, until we have fixed all of the commands (i.e. profile, token, publish, addUser) to be aware of this new config setup.

My suggestion then is to make each use of this an inline of this if statement, and we can optimize it later once we actually know the full story.

Reply:
Sure, that sounds good 👍